flytekit/core/data_persistence.py

```diff
@@ -438,7 +447,13 @@ def put_data(self, local_path: Union[str, os.PathLike], remote_path: str, is_mul
 DataPersistencePlugins.register_plugin("/", DiskPersistence)
 
 # TODO make this use tmpdir
-tmp_dir = os.path.join("/tmp/flyte", datetime.datetime.now().strftime("%Y%m%d_%H%M%S"))
+tmp_dir_prefix = "/tmp/flyte"
+
+# Update tmp_dir_prefix if running on Windows
+if os.name == "nt":
+    tmp_dir_prefix = os.path.join(tempfile.gettempdir(), "flyte")
+
+tmp_dir = os.path.join(tmp_dir_prefix, datetime.datetime.now().strftime("%Y%m%d_%H%M%S"))
```

Comment:
This is essentially creating a unique temp dir, right? We could do something like:
```
tmp_dir = tempfile.mkdtemp(prefix="/tmp/flyte")
```
So we wouldn't need to special-case windows (or any other OS fwiw).

Reply:
done, thanks